hw/top_earlgrey/data/clocks.xdc

```diff
@@ -18,9 +18,10 @@ set u_pll clkgen/pll
 set u_div2 top_*/u_clkmgr_aon/u_no_scan_io_div2_div
 create_generated_clock -name clk_io_div2 -source [get_pin ${u_pll}/CLKOUT0] -divide_by 2 [get_pin ${u_div2}/u_clk_div_buf/gen_xilinx.u_impl_xilinx/gen_fpga_buf.gen_bufg.bufg_i/O]
 
-set u_div4 top_*/u_clkmgr_aon/u_no_scan_io_div4_div
-create_generated_clock -name clk_io_div4 -source [get_pin ${u_pll}/CLKOUT0] -divide_by 4 [get_pin ${u_div4}/u_clk_div_buf/gen_xilinx.u_impl_xilinx/gen_fpga_buf.gen_bufg.bufg_i/O]
-
+# Recommended Div4 clock constraints from Xilinx
+# TODO uncomment below line and integrate u_div4 in path to make constraint more generic
+# set u_div4 top_*/u_clkmgr_aon/u_no_scan_io_div4_div
+create_generated_clock -name clkDiv4 -divide_by 4 -source [get_pins top_earlgrey/u_clkmgr_aon/u_no_scan_io_div4_div/gen_div.clk_int_reg/C] [get_pins top_earlgrey/u_clkmgr_aon/u_no_scan_io_div4_div/gen_div.clk_int_reg/Q]
```

Comment:
Please change the clock's name back to `clk_io_div4`. Other constraint files use the name, and a change now would be superfluous. (This is the reason for the build failures.)

Also, these are pin names that only come after synthesis of behavioral logic, and this can be finicky at times. I would prefer that we use names that explicitly exist in the source code and always survive synthesis renaming: We could reach the pins of primitives by traveling across the netlist instead (e.g. starting from the clock mux node's `clk0_i` pin and moving backwards to the driver pin, on the other side of the associated net). We can leave the bit in this paragraph for a later PR, though.

Reply:
Done.